chia/types/blockchain_format/proof_of_space.py

```diff
@@ -85,8 +85,8 @@ def passes_plot_filter(
     return cast(bool, plot_filter[:prefix_bits].uint == 0)
 
 
-def calculate_prefix_bits(constants: ConsensusConstants, height: uint32) -> int:
-    prefix_bits = constants.NUMBER_ZERO_BITS_PLOT_FILTER
+def calculate_prefix_bits(constants: ConsensusConstants, height: uint32) -> uint8:
+    prefix_bits: int = constants.NUMBER_ZERO_BITS_PLOT_FILTER
```

Comment:
?

```suggestion
    prefix_bits: uint8 = constants.NUMBER_ZERO_BITS_PLOT_FILTER
```

Reply:
It's possible for this to go negative, which is why there's a `max` at the end.

Reply:
Are we still working towards the fixed width types not subclassing `int`?  Though I guess if we are then a complaint would be made and addressed when we did that.

Reply:
I didn't know that was a goal, but that's out of scope for this PR either way

Reply:
One approach I was exploring in porting the fixed size int types to rust required first making them not inherit from `int`, because pyo3 doesn't support that. The new strategy is simpler. Any code written in rust will already use fixed size ints and I think it's fine if python code uses `int`

Reply:
Good point

Reply:
Do we need to convert it?  Remember that `uint8(5) + uint8(6)` gives you an `int` since we didn't set up any 'type retention' (i just made that term up, afaik) on these.

Reply:
right, I keep forgetting this. I think it's quite subtle, and causes `a += 1` to fail mypy (iirc) because `a` changes type.
I don't have a strong opinion, but I think it's quite reasonable to show that we want the `int` explicitly

Reply:
![image](https://github.com/Chia-Network/chia-blockchain/assets/543719/abec7927-49a3-4279-968f-944f6bcebc6d)

Quite so.  Alrighty, we can move forward.  Thanks.